src/key_grabber.c

```diff
@@ -148,14 +150,49 @@ void tilda_window_set_active (tilda_window *tw)
     DEBUG_FUNCTION ("tilda_window_set_active");
     DEBUG_ASSERT (tw != NULL);
 
+    gint mouse_x, mouse_y;
+    GdkDisplay *disp = gdk_display_get_default ();
+    GdkDeviceManager *device_manager = gdk_display_get_device_manager (disp);
+    GdkDevice *device = gdk_device_manager_get_client_pointer (device_manager);
+
+    /* Get the number of the monitor the mouse cursor is currently on */
     GdkScreen *screen = gtk_widget_get_screen (tw->window);
+    gdk_device_get_position (device, &screen, &mouse_x, &mouse_y);
+    gint mouse_monitor = gdk_screen_get_monitor_at_point (screen, mouse_x, mouse_y);
+
+    /* Get the monitor number on which the window is currently shown
+     * Idea: get the configured window position relatively to the configured
+     * monitor on which Tilda usually appears, and apply the same position
+     * on the monitor the mouse cursor is currently on */
+    int config_monitor = find_monitor_number (tw);
+    if (config_monitor != mouse_monitor) {
```

Comment:
This should be in a separate functions something like `int get_monitor_for_coursor(tw)` followed by `tilda_window_move_to_monitor (tw, monitorIndex)`.

Reply:
Done, this should be cleaner now.